splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceDateFunctions.java

```diff
@@ -456,17 +456,27 @@ public static double MONTH_BETWEEN(Date source1, Date source2) {
     /**
      * Implements the to_char function
      */
-    public static String TO_CHAR(Date source, String format) {
+    public static String TO_CHAR(Date source, String format) throws StandardException {
         if (source == null || format == null) return null;
 
+        try {
             SimpleDateFormat fmt = new SimpleDateFormat(format.toLowerCase().replaceAll("m", "M"));
             return fmt.format(source);
+        }
+        catch (Exception e) {
+            throw StandardException.newException(SQLState.LANG_FORMAT_EXCEPTION, "datetime");
+        }
 
     }
-    public static String TIMESTAMP_TO_CHAR(Timestamp stamp, String output) {
+    public static String TIMESTAMP_TO_CHAR(Timestamp stamp, String output) throws StandardException {
         if (stamp == null || output == null) return null;
-        SimpleDateFormat fmt = new SimpleDateFormat(output.toLowerCase().replaceAll("m", "M"));
-        return fmt.format(stamp);
+        try {
+            SimpleDateFormat fmt = new SimpleDateFormat(output.toLowerCase().replaceAll("m", "M"));
```

Comment:
This is unrelated to your change, but the `.toLowerCase()` call seems wrong. There are case sensitive format patterns (a/A, k/K, e/E...) so we are losing that information. If it's as simple as removing the call and all ITs work can you fix it? Otherwise please open a new Jira.

Reply:
one It failed to parse a date the correct format.

Reply:
@mush-zhang what IT?

Reply:
testTimestampToChar

Reply:
> @mush-zhang what IT?

testTimestampToChar in SpliceDateFunctionIT

Reply:
Ok, I opened https://splicemachine.atlassian.net/browse/DB-11882